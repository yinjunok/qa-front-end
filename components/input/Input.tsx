import * as React from 'react';
import * as css from './styles.less';

interface ISearchBox extends React.InputHTMLAttributes<HTMLInputElement> {
  rightIcon: React.ReactNode | null;
  onFocus?: (e: React.FocusEvent<HTMLInputElement>) => void;
  onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
}

export default class SearchBox extends React.Component<ISearchBox, {}> {
  state = {
    isFocus: false,
  }

  static defaultProps = {
    rightIcon: null,
    onFocus: () => {},
    onBlur: () => {}
  }

  render() {
    const {
      rightIcon,
      className,
      onFocus,
      onBlur,
      ...props
    } = this.props;
    const { isFocus } = this.state;

    return (
      <div className={`${css.inputWrapper} ${isFocus ? css.inputActive : ''}`}>
        <input
          className={css.input}
          onFocus={(e: React.FocusEvent<HTMLInputElement>) => this.toggleFocusHandler(e)}
          onBlur={(e: React.FocusEvent<HTMLInputElement>) => this.toggleFocusHandler(e)}
          {...props}
        />
        {rightIcon}
      </div>
    );
  }

  private toggleFocusHandler = (e: React.FocusEvent<HTMLInputElement>) => {
    this.setState({
      isFocus: !this.state.isFocus,
    });
   
    if (e.type = 'focus') {
      this.setState({
        isFocus: !this.state.isFocus,
      });
      const { onFocus } = this.props;
      onFocus && onFocus(e);
    }

    if (e.type = 'blur') {
      const { onBlur } = this.props;
      onBlur && onBlur(e);
    }
  }
}
