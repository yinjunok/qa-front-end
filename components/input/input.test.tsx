import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Input from './Input';

describe('snapshot', () => {
  test('no rightIcon', () => {
    const input = renderer.create(<Input />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test('right Icon', () => {
    const input = renderer.create(<Input rightIcon={<button>按钮</button>} />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test('with className', () => {
    const input = renderer.create(<Input className='className' />);
    expect(input.toJSON()).toMatchSnapshot();
  });
});

describe('function call', () => {
  test('onFocus', () => {
    const onFocus = jest.fn();
    const input = mount(<Input onFocus={onFocus} />);
    input.find('input').simulate('focus', onFocus);
    expect(onFocus).toBeCalled();

    expect(input.state('isFocus')).toBe(true);
  });

  test('onBlur', () => {
    const cb = jest.fn();
    const input = mount(<Input onFocus={cb} />);

    input.find('input').simulate('focus');
    input.find('input').simulate('blur', cb);
    expect(cb).toBeCalled();

    expect(input.state('isFocus')).toBe(false);
  });
});