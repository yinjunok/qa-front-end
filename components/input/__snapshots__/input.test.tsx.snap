// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot no rightIcon 1`] = `
<div
  className="inputWrapper "
>
  <input
    className="input"
    onBlur={[Function]}
    onFocus={[Function]}
  />
</div>
`;

exports[`snapshot right Icon 1`] = `
<div
  className="inputWrapper "
>
  <input
    className="input"
    onBlur={[Function]}
    onFocus={[Function]}
  />
  <button>
    按钮
  </button>
</div>
`;

exports[`snapshot with className 1`] = `
<div
  className="inputWrapper "
>
  <input
    className="input"
    onBlur={[Function]}
    onFocus={[Function]}
  />
</div>
`;
